piksi_tools/settings.py

```diff
@@ -77,7 +78,6 @@ def __enter__(self):
                                SBP_MSG_SETTINGS_READ_BY_INDEX_RESP)
         self.link.add_callback(self._settings_done_callback,
                                SBP_MSG_SETTINGS_READ_BY_INDEX_DONE)
-        self.link.add_callback(self._print_callback, [SBP_MSG_LOG])
```

Comment:
removing the SBP_MSG_LOG callback may lose a human friendly textual hint from the firmware about why a settings write was rejected.  Do we think the return codes cover all the hints that are required?

Reply:
I think it works otherwise well except it lacks the "NTRIP must be disabled to modify settings" message, giving just a more generic "setting is read-only" error. However, this depends on new firmware.